dot/parachain/candidate-validation/worker.go

```diff
@@ -31,7 +35,30 @@ func newWorker(validationCode parachaintypes.ValidationCode) (*worker, error) {
 	}, nil
 }
 
+type resultWithError struct {
+	result *parachainruntime.ValidationResult
+	err    error
+}
+
+func determineTimeout(timeoutKind parachaintypes.PvfExecTimeoutKind) time.Duration {
+	value, err := timeoutKind.Value()
+	if err != nil {
+		return 2 * time.Second
+	}
+	switch value.(type) {
+	case parachaintypes.Backing:
+		return 2 * time.Second
+	case parachaintypes.Approval:
+		return 12 * time.Second
+	default:
+		return 2 * time.Second
```

Comment:
just a simplification
```suggestion
	case parachaintypes.Approval:
		return 12 * time.Second
	default:
		return 2 * time.Second
```

Reply:
Updated.